### PR TITLE
fix: respect `ssr.resolve.conditions` on ssr environment

### LINKED ISSUE #6992: Don't respect `conditions` on Vite 6
### Describe the bug

- `resolve.conditions` is not respected on Vite 6 with Vitest.
- It works on Vite 6 without Vitest.
- It works on Vite 5.

```ts
// vitest.config.ts
import { defineConfig } from 'vitest/config'

export default defineConfig({
  resolve: {
    conditions: ['dev'],
  },
})
```

### Reproduction

https://stackblitz.com/~/github.com/sxzz/vitest-conditions-repro

- Run `pnpm run test`, and you will see the test has failed.
- Downgrade to vite@5, remove `node_modules` and `pnpm-lock.yaml`, and it works.

### System Info

```shell
N/A
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by sxzz**
### Description

Respect `resolve.conditions` for ssr env.

closes #6992

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/vitest/src/node/plugins/index.ts [hunk 1]
```diff
@@ -121,7 +121,7 @@ export async function VitestPlugin(
                 // by default Vite resolves `module` field, which not always a native ESM module
                 // setting this option can bypass that and fallback to cjs version
                 mainFields: [],
```
Comment (sxzz) [inline packages/vitest/src/node/plugins/index.ts:123]: No idea about `mainFields`. Should we respect it from user config as well?

```suggestion
                mainFields: viteConfig.resolve?.mainFields || [],
```

**Comment by sheremet-va** [comment]
As I said in https://github.com/vitest-dev/vitest/issues/6992#issuecomment-2509454156 this is expected breaking change in Vite

**Comment by sxzz** [comment]
@sheremet-va I just updated it to `ssr.resolve.conditions`.

**Comment by hi-ogawa** [comment]
Closing since this compat is implemented by Vite https://github.com/vitejs/vite/pull/18857

**Pull request closed without merging**